JetStreamCompose/jetstream/src/main/java/com/google/jetstream/data/repositories/MovieRepositoryImpl.kt

```diff
@@ -33,132 +33,167 @@ import com.google.jetstream.data.util.StringConstants.Movie.Reviewer.DefaultCoun
 import com.google.jetstream.data.util.StringConstants.Movie.Reviewer.DefaultRating
 import com.google.jetstream.data.util.StringConstants.Movie.Reviewer.FreshTomatoes
 import com.google.jetstream.data.util.StringConstants.Movie.Reviewer.ReviewerName
+import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.flow.flow
-import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.withLock
+import kotlinx.coroutines.withContext
 import kotlinx.serialization.json.Json
 import javax.inject.Inject
 
-class MovieRepositoryImpl @Inject constructor(assetsReader: AssetsReader) : MovieRepository {
+class MovieRepositoryImpl @Inject constructor(
```

Comment:
because you're using some caching mechanisms, I assume you would like this repository to be stateful and thus use `@Singleton`?

Reply:
Yes. That is my intention.

Reply:
```suggestion
@Singleton
class MovieRepositoryImpl @Inject constructor(
```